elasticsearch/run.sh

```diff
@@ -9,20 +9,52 @@ secret_dir=/etc/elasticsearch/secret/
 [ -f $secret_dir/keystore.password ] && export KEYSTORE_PASSWORD=$(cat $secret_dir/keystore.password)
 [ -f $secret_dir/truststore.password ] && export TRUSTSTORE_PASSWORD=$(cat $secret_dir/truststore.password)
 
+BYTES_PER_MEG=$((1024*1024))
+BYTES_PER_GIG=$((1024*${BYTES_PER_MEG}))
+
+MAX_ES_MEMORY_BYTES=$((64*${BYTES_PER_GIG}))
+MIN_ES_MEMORY_BYTES=$((256*${BYTES_PER_MEG}))
+
 # the amount of RAM allocated should be half of available instance RAM.
 # ref. https://www.elastic.co/guide/en/elasticsearch/guide/current/heap-sizing.html#_give_half_your_memory_to_lucene
+# parts inspired by https://github.com/fabric8io-images/run-java-sh/blob/master/fish-pepper/run-java-sh/fp-files/java-container-options
 regex='^([[:digit:]]+)([GgMm])$'
 if [[ "${INSTANCE_RAM:-}" =~ $regex ]]; then
 	num=${BASH_REMATCH[1]}
 	unit=${BASH_REMATCH[2]}
 	if [[ $unit =~ [Gg] ]]; then
-		((num = num * 1024)) # enables math to work out for odd Gi
+		((num = num * ${BYTES_PER_GIG})) # enables math to work out for odd Gi
+	elif [[ $unit =~ [Mm] ]]; then
```

Comment:
I don't really follow since this is still required to properly convert to the base unit of bytes.